src/dbnode/ts/segment.go

```diff
@@ -129,29 +129,36 @@ func (s *Segment) Finalize() {
 func (s *Segment) Clone(pool pool.CheckedBytesPool) Segment {
 	var (
 		checkedHead, checkedTail checked.Bytes
+		tail                     []byte
```

Comment:
@robskillington I also changed this segment `Clone()` code to account for the case where the `Tail` is `nil`.

Reply:
Yeah good call, we could even update it to check if head is nil or not nil too lol, but maybe overkill.